lib/backend/backend.go

```diff
@@ -244,6 +245,14 @@ func NextPaginationKey(r types.Resource) string {
 	return string(nextKey([]byte(r.GetName())))
 }
 
+// MaskKeyName masks the given key name.
+// e.g "123456789" -> "******789"
+func MaskKeyName(str string) string {
```

Comment:
This seems pretty general purpose and not specific to the backend - is there a better package we could put this logic in?

Reply:
I put it here because right now it is only being used to mask backend key names. This could be moved to a new file in `lib/utils`, but our utils are already overflowing and I think it's better to keep this here until (if) it is needed for more general use cases.